src/engine/GroupBy.cpp

```diff
@@ -982,9 +1174,12 @@ void GroupBy::substituteAllAggregates(
 
     AD_CONTRACT_CHECK(aggregate.parentAndIndex_.has_value());
     auto parentAndIndex = aggregate.parentAndIndex_.value();
+    originalChildren.push_back(std::move(
+        parentAndIndex.parent_->children()[parentAndIndex.nThChild_]));
     parentAndIndex.parent_->replaceChild(parentAndIndex.nThChild_,
```

Comment:
What about
`originalChildren.push_back(parentAndI.parent_->replaceChild(...))`
(  Let `replaceChild` return the moved out child, makes the code much simpler here.
The change in `replaceChild` can probably be implemented using a single line via `std::exchange`.